js/main.js

```diff
@@ -5,13 +5,20 @@ function getCryptoData(id) {
   xhr.addEventListener('load', function () {
     $detailsPicture.setAttribute('src', xhr.response[0].image);
     $detailsName.textContent = xhr.response[0].name + ' ' + '(' + xhr.response[0].symbol.toUpperCase() + ')';
-    $price.textContent = 'Price: ' + xhr.response[0].current_price;
-    $marketcap.textContent = 'Market Cap: ' + xhr.response[0].market_cap;
+    $price.textContent = 'Price: ' + '$' + xhr.response[0].current_price;
+    $marketcap.textContent = 'Market Cap: ' + '$' + xhr.response[0].market_cap;
     $rank.textContent = 'Rank: ' + xhr.response[0].market_cap_rank;
     $searchView.className = 'search-view hidden';
     $detailsView.className = 'details-view-gray-background';
     data.view = 'details';
   });
+  var $addButton = document.querySelector('.add-button');
+  $addButton.addEventListener('click', addHandler);
+
+  function addHandler(event) {
+    data.list.push(xhr.response[0]);
+
+  }
```

Comment:
It would be best if you did not have functions inside of other functions. I know this is because you need access to the XHR response, but instead, you should save the data for the currently visible currency in your data model (maybe a property named `currentCurrencyData`) and then when the user clicks the button to add to their watchlist, you should use the data in that property. That will allow you to move this function out to the global scope, as it should be.